exporters/memory/test/in_memory_span_data_test.cc

```diff
@@ -0,0 +1,33 @@
+#include "opentelemetry/exporters/memory/in_memory_span_data.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+#include <gtest/gtest.h>
+
+using opentelemetry::exporter::memory::InMemorySpanData;
+using opentelemetry::sdk::trace::Recordable;
+using opentelemetry::sdk::trace::SpanData;
+
+TEST(InMemorySpanData, AddRecordable)
+{
+  InMemorySpanData data(100);
+
+  ASSERT_EQ(0, data.GetSpans().size());
+
+  opentelemetry::nostd::span<std::unique_ptr<Recordable>> recordables(
+      new std::unique_ptr<Recordable>(new SpanData), 1);
+
+  for (auto &recordable : recordables)
+  {
+    auto span = std::unique_ptr<SpanData>(static_cast<SpanData *>(recordable.release()));
+    if (span != nullptr)
+    {
+      data.Add(std::move(span));
+    }
+  }
+
+  ASSERT_EQ(1, data.GetSpans().size());
+
+  // Consumes all spans in exporter
```

Comment:
nit: this comment is for the first ASSERT right?

Reply:
Correct, I've went ahead and moved it to clear up any confusion